go/client/cmd_ctl_start.go

```diff
@@ -4,33 +4,73 @@
 package client
 
 import (
+	"fmt"
+
 	"github.com/keybase/cli"
+	"github.com/keybase/client/go/install"
 	"github.com/keybase/client/go/libcmdline"
 	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
 )
 
 func NewCmdCtlStart(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
 	return cli.Command{
 		Name:  "start",
 		Usage: "Start the background keybase service",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "t, type",
+				Usage: fmt.Sprintf("Service type (%s, %s), default is %q", keybase1.ForkType_AUTO.String(), keybase1.ForkType_LAUNCHD.String(), defaultForkType.String()),
+			},
+		},
 		Action: func(c *cli.Context) {
-			cl.ChooseCommand(&CmdCtlStart{libkb.NewContextified(g)}, "start", c)
-			cl.SetForkCmd(libcmdline.ForceFork)
-			cl.SetNoStandalone()
+			cl.ChooseCommand(NewCmdCtlStartRunner(g), "start", c)
+			cl.SetForkCmd(libcmdline.NoFork)
+			cl.SetLogForward(libcmdline.LogForwardNone)
 		},
 	}
 }
 
 type CmdCtlStart struct {
 	libkb.Contextified
+	forkType keybase1.ForkType
+}
+
+func NewCmdCtlStartRunner(g *libkb.GlobalContext) *CmdCtlStart {
+	return &CmdCtlStart{Contextified: libkb.NewContextified(g)}
 }
 
 func (s *CmdCtlStart) ParseArgv(ctx *cli.Context) error {
+	forkTypeStr := ctx.String("type")
```

Comment:
I don't think we should waste any time changing and/or debugging things
here  that already work.  We should keep them warts and all just so we can
focus on more important things.

How about these proposed fixes:
# keep ctl start exactly as it was before for non OSX
# for OSX, have it autofork if --auto-fork was specified and have it
launchd start otherwise. This way we don't need any new flags.

Is this ok ?

On Monday, March 21, 2016, Gabriel Handford <notifications@github.com>
wrote:

> In go/client/cmd_ctl_start.go
> <https://github.com/keybase/client/pull/2414#discussion_r56903347>:
>
> >  }
> >
> >  func (s *CmdCtlStart) ParseArgv(ctx *cli.Context) error {
> > +	forkTypeStr := ctx.String("type")
>
> Maybe for backwards compatibility?
> Otherwise this totally should be an enum... you can't fork multiple ways :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/2414/files/25717f00082457bb975b8db16f245eeeda3bde96#r56903347>
>